### DISCUSSION TITLE
Deleting media doesn't remove from the current view

### DISCUSSION THREAD

**Discussion by brianjmurrell** [jellyfin/jellyfin#10700] [General]
**Describe the bug**
If I'm on the page for a given show, listing the episodes and I delete an episode, the removal is not reflected on that page until I back out of it and then go back into it.

**To Reproduce**
1. Go to a TV show page where you have episodes.
1. On an episodes's menu select "Delete Media"

**Expected behavior**
The deleted item should be removed from the visible page immediately.

**System (please complete the following information):**
 - OS: Fedora 29
 - Browser: Chrome
 - Jellyfin Version: 10.1.0

**Comment by stale** [comment]
Issues go stale after 60d of inactivity. Mark the issue as fresh by adding a comment or commit. Stale issues close after an additional 7d of inactivity. If this issue is safe to close now please do so. If you have any questions you can reach us on [Matrix or Social Media](https://jellyfin.readthedocs.io/en/latest/getting-help/).

**Comment by brianjmurrell** [comment]
This is still an issue.

**Comment by stale** [comment]
Issues go stale after 90d of inactivity. Mark the issue as fresh by adding a comment or commit. Stale issues close after an additional 14d of inactivity. If this issue is safe to close now please do so. If you have any questions you can reach us on [Matrix or Social Media](https://docs.jellyfin.org/general/getting-help.html).

**Comment by brianjmurrell** [comment]
Go away @stalebot.

Can we disable stalebot for **real** known bugs please?  Like exclude issues that have the label **bug**.

**Comment by dkanada** [comment]
I'm inclined to agree.

**Comment by classicjazz** [comment]
I believe the reason for this is two different issues:

1.) the web page may not implement auto refresh after deleting a video

2.) "delete media" deletes the *.ts, *.jpg and *.nfo corresponding to episodes but leaves the folder structure and tvshow.nfo intact. If there are no more videos remaining for a show, the folder structure and tvshow.nfo should be deleted too. Then, the TV show will no longer appear on the Shows page.

To validate this, you can compare the difference between using the "delete media" menu item versus manually deleting a TV show folder at the file system level. A file system delete (e.g. using Synology DSM) results in the TV show being removed from the JellyFin UI.

**Comment by brianjmurrell** [comment]
@classicjazz I don't think either of your explanations are valid.

1. Auto-refresh is not needed as the page should be specifically updated to reflect the new state when the webui transaction to do the delete is complete.
1. Is orthogonal to the issue being described here.  The issue being describe here is specifically about deleted episodes, not shows and their metadata.

> To validate this, you can compare the difference between using the "delete media" menu item versus manually deleting a TV show folder at the file system level. A file system delete (e.g. using Synology DSM) results in the TV show being removed from the JellyFin UI.

Without confirming this behaviour[1], if it is true, then it makes the lack of updating of the page with the show listed that you just deleted (by the menu) even worse.  If the page can dynamically update when the filesystem is modified underneath it, it surely must be able to be updated when it knows exactly what the update (a delete) was.

[1] I tried this process out but the web UI did not update even though Jellyfin did update (per the log).  I suspect because as you suggest, there is no auto-update or AJAX or any such thing to cause the page to be refreshed.  Manually refreshing did update the page with the episode deleted from the filesystem though.

**Comment by MCBuckethead** [comment]
I took a look at this to see if it's still around and it is. I did see, however, that if you were to delete a movie or photo the page refreshes as expected.

**Comment by sevenrats** [comment]
>  The issue being describe here is specifically about deleted episodes, not shows and their metadata.
 
if so, please change the title to reflect this.

**Comment by brianjmurrell** [comment]
I think it's orthogonal.  Ultimately whatever you delete whether it's a single episode, a season or a whole show, the UI should auto-refresh to reflect what was deleted.

**Comment by sevenrats** [comment]
@brianjmurrell l this is not so much a bug report though as a long-form discussion of an eventual feature.  do you have any objection to it being converted to a discussion? lots of stuff in jelly land doesn't refresh the UI as you might expect, and a place to discuss these seems appropriate to me.

**Comment by brianjmurrell** [comment]
Given that this issue has been open for nearly 5 years, I probably don't really care what happens with it at this point.  Whatever you do with it, please at least post a link here to wherever it gets moved to.

**Comment by brianjmurrell** [comment]
I know I said I don't really care, but what is there really to _discuss_ about this?  The issue (when it was an issue) was quite clear.  The UI does not immediately (i.e. without manual refreshes) reflect the remote state of deleted items.

Now that I see what discussion item looks like, this does not seem an appropriate place for this issue.

Discussions appear to be more of a _question and answer_ forum.  There is no answer for this issue.  It's a defect/shortcoming of the software that needs fixing.  This is more appropriately an issue, not a discussion IMO.

**Reply by sevenrats** [reply]
well its not a bug in jellyfin terms because there is no code that is supposed to do this.  and even if we were to *treat* it as a bug its not specific enough to close with a particular patch or pull.  plus its not immediately clear whether it will be solved with autorefresh, or another mechanism.  all these things can be discussed here.

**Comment by TheMelmacian** [comment]
Since this is an issue with the web-frontend and not the server, shouldn't this issue/discussion be moved to the https://github.com/jellyfin/jellyfin-web repo?

**Reply by sevenrats** [reply]
to me its like error codes where both server and client are involved, and, because it could potentially affect multiple clients, it makes most sense to me to keep the discussion here, which is central to all the points the issue may relate to.

**Reply by sevenrats** [reply]
i could be convinced that it should be moved to meta, but, really lots of meta discussions happen in the server repo.

**Reply by TheMelmacian** [reply]
OK, I get your point. But the handling has to be done on the client side. The server responds with a 200 or 204 code, if the deletion was successful, and the client should react to it.

For the web client I can think of two ways: the already mentioned page refresh (sufficient but a bit hacky, because the whole page and all elements have to be reloaded), or the client must delete the element from the list and update the view (IMHO the better way to do it).
Clients that don't use the standard web-ui must implement their own handling.

**Comment by thornbill** [comment]
Agreed this really is on the clients to sort out.

For jellyfin-web specifically, the old view code really doesn't have a good way to update information dynamically without a full page refresh. There are a couple places where we _try_ to refresh things but it's a pain to manage.

Going forward this will be _much_ easier though. Using react with react-query, we can pretty easily have data displayed in the ui refreshed on a regular interval and immediately invalidated when an edit is made. There is still a lot of work to do to get things converted over, but we have the tools we need and work is underway.

**Comment by sevenrats** [comment]
@thornbill is this polling behavior the only way to achieve this goal on other clients?  seems to me that thats going to be a non-trivial burden in some of them.

**Reply by thornbill** [reply]
Yes, mostly. The only alternative I can think of would be to use a websocket message to notify clients that they need to update or provide updated data directly... but that seems much more complicated and would still be a lot of work for clients to add support. 🤷‍♂️
